internal/services/automation/automation_runtime_environment_resource.go

```diff
@@ -0,0 +1,289 @@
+		},
+
+		"resource_group_name": commonschema.ResourceGroupName(),
+
+		"automation_account_name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validate.AutomationAccount(),
+		},
+
+		"runtime_language": {
+			Type:     pluginsdk.TypeString,
+			Required: true,
+			ForceNew: true,
+			ValidateFunc: validation.StringInSlice([]string{
+				string("Python"),
+				string("PowerShell"),
+			}, false),
+		},
+
+		"runtime_version": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"runtime_default_packages": {
+			Type:     pluginsdk.TypeMap,
+			Optional: true,
+			Elem:     &pluginsdk.Schema{Type: pluginsdk.TypeString},
+		},
+
+		"location": commonschema.Location(),
+
+		"description": {
+			Type:         pluginsdk.TypeString,
+			Optional:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"tags": commonschema.Tags(),
+	}
+}
+
+func (m AutomationRuntimeEnvironmentResource) Attributes() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{}
+}
+
+func (m AutomationRuntimeEnvironmentResource) ModelObject() interface{} {
+	return &AutomationRuntimeEnvironmentResourceModel{}
+}
+
+func (m AutomationRuntimeEnvironmentResource) ResourceType() string {
+	return "azurerm_automation_runtime_environment"
+}
+
+func (m AutomationRuntimeEnvironmentResource) Create() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 10 * time.Minute,
+		Func: func(ctx context.Context, meta sdk.ResourceMetaData) error {
+			client := meta.Client.Automation.RuntimeEnvironment
+
+			var model AutomationRuntimeEnvironmentResourceModel
+			if err := meta.Decode(&model); err != nil {
+				return fmt.Errorf("decoding: %+v", err)
+			}
+
+			subscriptionID := meta.Client.Account.SubscriptionId
+			id := runtimeenvironment.NewRuntimeEnvironmentID(subscriptionID, model.ResourceGroupName, model.AutomationAccountName, model.Name)
+
+			existing, err := client.Get(ctx, id)
+			if err != nil && !response.WasNotFound(existing.HttpResponse) {
+				return fmt.Errorf("checking for presence of existing %s: %+v", id, err)
+			}
+
+			if !response.WasNotFound(existing.HttpResponse) {
+				return meta.ResourceRequiresImport(m.ResourceType(), id)
+			}
+
+			req := runtimeenvironment.RuntimeEnvironment{}
+
+			req.Properties = &runtimeenvironment.RuntimeEnvironmentProperties{
+				Runtime: &runtimeenvironment.RuntimeProperties{
+					Language: pointer.FromString(model.RuntimeLanguage),
+					Version:  pointer.FromString(model.RuntimeVersion),
+				},
+				DefaultPackages: &model.RuntimeDefaultPackages,
+				Description:     pointer.FromString(model.Description),
+			}
+
+			req.Location = azure.NormalizeLocation(model.Location)
+			req.Tags = &model.Tags
+
+			if _, err = client.Create(ctx, id, req); err != nil {
+				return fmt.Errorf("creating %s: %v", id, err)
```

Comment:
We should use %+v for verbose error formatting to provide full error context. Same for the other error messages in this file.

Reply:
Lines 150, 172, 198, 260, 280: Error messages use %v instead of %+v